mmgen/models/architectures/sngan_proj/modules.py

```diff
@@ -0,0 +1,236 @@
+                 shortcut_cfg=None,
+                 act_cfg=dict(type='ReLU')):
+
+        super().__init__()
+        self.activate = build_activation_layer(act_cfg)
+        conv_blocks = [
+            ConvModule(in_channels, out_channels, **conv_cfg),
+            ConvModule(out_channels, out_channels, **conv_cfg)
+        ]
+        self.conv_blocks = nn.ModuleList(conv_blocks)
+
+        self.downsample = nn.AvgPool2d(2, 2)
+
+        self.shortcut = ConvModule(in_channels, out_channels, **shortcut_cfg)
+
+    def forward(self, x):
+        out = self.conv_blocks[0](x)
+        out = self.activate(out)
+        out = self.conv_blocks[1](out)
+        out = self.downsample(out)
+
+        shortcut = self.forward_shortcut(x)
+        return out + shortcut
+
+    def forward_shortcut(self, x):
+        out = self.downsample(x)
+        out = self.shortcut(out)
+        return out
+
+    def _weight_init(self):
+        xavier_init(self.conv, gain=np.sqrt(2), distribution='uniform')
+        if self.learnable_sc:
+            xavier_init(self.shortcut, gain=1, distribution='uniform')
+
+
+@MODULES.register_module()
+class SNcBatchNorm(nn.Module):
+
+    def __init__(self,
+                 in_channels,
+                 num_classes,
+                 use_cbn=True,
+                 norm_cfg=dict(type='BN'),
+                 use_norm_affine=False,
+                 auto_sync_bn=True):
+        super().__init__()
+        norm_cfg = deepcopy(norm_cfg)
+        norm_type = norm_cfg['type']
+
+        if norm_type not in ['IN', 'BN', 'syncBN']:
+            raise ValueError('Only support `IN` (InstanceNorm), '
+                             '`BN` (BatcnNorm) and `syncBN` for '
+                             'Class-conditional bn. '
+                             'Receive norm_type: {}'.format(norm_type))
+        norm_cfg['affine'] = use_norm_affine
+        _, self.norm = build_norm_layer(norm_cfg, in_channels)
+
+        if auto_sync_bn and check_dist_init():
+            from torch.nn.modules.batchnorm import SyncBatchNorm
+            self.norm = SyncBatchNorm.convert_sync_batchnorm(self.norm)
+
+        self.use_cbn = use_cbn
+        if use_cbn:
+            if num_classes <= 0:
+                raise ValueError('`num_classes` must be larger '
+                                 'than 0 with `use_cbn=True`')
+            self.weight_embedding = nn.Embedding(num_classes, in_channels)
+            self.biase_embedding = nn.Embedding(num_classes, in_channels)
+        self._init_weight()
+
+    def forward(self, x, y=None):
+        out = self.norm(x)
+        if self.use_cbn:
+            weight = self.weight_embedding(y).view(y.size(0), -1, 1, 1)
+            bias = self.biase_embedding(y).view(y.size(0), -1, 1, 1)
+            out = out * weight / bias
+        return out
+
+    def _init_weight(self):
+        if self.use_cbn:
+            constant_init(self.weight_embedding, 1)
+            constant_init(self.biase_embedding, 1)
```

Comment:
Is this initialization right? In general, bias tensor should be initialized as `0`.